#!/usr/bin/env python
PACKAGE = "reflex_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("translational_stiffness", double_t, 0, "Cartesian translational stiffness", 200, 0, 400)
gen.add("rotational_stiffness", double_t, 0, "Cartesian rotational stiffness", 10, 0, 30)
gen.add("nullspace_stiffness", double_t, 0, "Stiffness of the joint space nullspace controller (the desired configuration is the one at startup)", 0, 0, 100)

gen.add("gravReflex_translational_stiffness", double_t, 0, "Cartesian translational stiffness", 0, 0, 400)
gen.add("gravReflex_rotational_stiffness", double_t, 0, "Cartesian rotational stiffness", 0, 0, 30)
gen.add("gravReflex_nullspace_stiffness", double_t, 0, "Stiffness of the joint space nullspace controller", 0, 0, 100)

gen.add("impReflex_translational_stiffness", double_t, 0, "Cartesian translational stiffness", 200, 0, 400)
gen.add("impReflex_rotational_stiffness", double_t, 0, "Cartesian rotational stiffness", 10, 0, 30)
gen.add("impReflex_nullspace_stiffness", double_t, 0, "Stiffness of the joint space nullspace controller", 0, 0, 100)

gen.add("cptc_translational_stiffness", double_t, 0, "Cartesian translational stiffness", 200, 0, 400)
gen.add("cptc_rotational_stiffness", double_t, 0, "Cartesian rotational stiffness", 10, 0, 30)
gen.add("cptc_nullspace_stiffness", double_t, 0, "Stiffness of the joint space nullspace controller", 0, 0, 100)
gen.add("cptc_vel", double_t, 0, "Cartesian velocity ", 0.3, 0, 10)
gen.add("cptc_vel_rotation", double_t, 0, "Cartesian rotation velocity", 0.5, 0, 30)
gen.add("cptc_tol_trans", double_t, 0, "Pose tolerance", 0.05, 0, 100)
gen.add("cptc_tol_rot", double_t, 0, "Angle tolerance", 0.05, 0, 100)

exit(gen.generate(PACKAGE, "dynamic_compliance", "compliance_param"))
